test/integration/hermetic_test.go

```diff
@@ -397,23 +397,42 @@ func setUpHermeticServer(podMetrics []*datastore.PodMetrics) (client extProcPb.E
 	pmc := &backend.FakePodMetricsClient{Res: pms}
 
 	serverCtx, stopServer := context.WithCancel(context.Background())
-	go func() {
-		serverRunner.Datastore.PodDeleteAll()
-		for _, pm := range podMetrics {
-			pod := utiltesting.MakePod(pm.NamespacedName.Name).
-				Namespace(pm.NamespacedName.Namespace).
-				ReadyCondition().
-				IP(pm.Address).
-				ObjRef()
-			serverRunner.Datastore.PodUpdateOrAddIfNotExist(pod)
-			serverRunner.Datastore.PodUpdateMetricsIfExist(pm.NamespacedName, &pm.Metrics)
+
+	// TODO: this should be consistent with the inference pool
+	podLabels := map[string]string{
+		"app": "vllm-llama2-7b-pool",
+	}
+
+	for _, pm := range podMetrics {
+		pod := utiltesting.MakePod(pm.NamespacedName.Name).
+			Namespace(pm.NamespacedName.Namespace).
+			ReadyCondition().
+			Labels(podLabels).
+			IP(pm.Address).
+			Complete().
+			ObjRef()
+
+		copy := pod.DeepCopy()
+		if err := k8sClient.Create(context.Background(), copy); err != nil {
+			logutil.Fatal(logger, err, "Failed to create pod", "pod", pm.NamespacedName)
+		}
+
+		// since no pod controllers deployed in fake environment, we manually update pod status
+		copy.Status = pod.Status
+		if err := k8sClient.Status().Update(context.Background(), copy); err != nil {
+			logutil.Fatal(logger, err, "Failed to update pod status", "pod", pm.NamespacedName)
 		}
-		serverRunner.Provider = backend.NewProvider(pmc, serverRunner.Datastore)
+	}
+	serverRunner.Provider = backend.NewProvider(pmc, serverRunner.Datastore)
+	go func() {
 		if err := serverRunner.AsRunnable(logger.WithName("ext-proc")).Start(serverCtx); err != nil {
 			logutil.Fatal(logger, err, "Failed to start ext-proc server")
 		}
 	}()
 
+	// sleep 5 seconds to wait for datastore to be synced
+	time.Sleep(5 * time.Second)
```

Comment:
I suggest we avoid setting random sleeps like this, and use something like Eventually: https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/main/test/integration/hermetic_test.go#L522.

This will be alot easier if we migrate to use ginkgo!